smack-experimental/src/main/java/org/jivesoftware/smackx/mam/MamManager.java

```diff
@@ -348,39 +346,24 @@ private MamQueryIQ prepareMamQueryIQGet(String queryId) {
         return mamQueryIQ;
     }
 
-    private MamQueryResult queryArchive(MamQueryIQ mamQueryIq, long extraTimeout) throws NoResponseException,
-            XMPPErrorException, NotConnectedException, InterruptedException, NotLoggedInException {
-        if (extraTimeout < 0) {
-            throw new IllegalArgumentException("extra timeout must be zero or positive");
-        }
-
+    private MamQueryResult queryArchive(MamQueryIQ mamQueryIq) throws NoResponseException, XMPPErrorException,
+            NotConnectedException, InterruptedException, NotLoggedInException {
         final XMPPConnection connection = getAuthenticatedConnectionOrThrow();
-        MamFinIQ mamFinIQ = null;
-
-        PacketCollector finIQCollector = connection.createPacketCollector(new MamFinIQFilter(mamQueryIq));
 
         PacketCollector.Configuration resultCollectorConfiguration = PacketCollector.newConfiguration()
-                .setStanzaFilter(new MamResultFilter(mamQueryIq)).setCollectorToReset(finIQCollector);
-
+                .setStanzaFilter(new MamResultFilter(mamQueryIq));
         PacketCollector resultCollector = connection.createPacketCollector(resultCollectorConfiguration);
 
-        try {
-            connection.createPacketCollectorAndSend(mamQueryIq).nextResultOrThrow();
-            IQ iq = finIQCollector.nextResultOrThrow(connection.getPacketReplyTimeout() + extraTimeout);
-            mamFinIQ = MamFinIQ.from(iq);
-        } finally {
-            resultCollector.cancel();
-            finIQCollector.cancel();
-        }
```

Comment:
I wonder why you changed this method? Now the packet collectors logic is broken. You need to ensure the *both* collectors are canceled in *any* case. My original code ensured that. Your new code does not. You also need to use `setCollectorToReset()` to reset the timeout counter.